apps/meteor/ee/server/api/engagementDashboard/channels.ts

```diff
@@ -45,17 +46,28 @@ API.v1.addRoute(
 				Match.ObjectIncluding({
 					start: Match.Where(isDateISOString),
 					end: Match.Where(isDateISOString),
+					hideRoomsWithNoActivity: Match.Maybe(String),
 					offset: Match.Maybe(String),
 					count: Match.Maybe(String),
 				}),
 			);
 
-			const { start, end } = this.queryParams;
+			const { start, end, hideRoomsWithNoActivity } = this.queryParams;
 			const { offset, count } = await getPaginationItems(this.queryParams);
 
-			const { channels, total } = await findAllChannelsWithNumberOfMessages({
+			apiDeprecationLogger.deprecatedParameterUsage(
```

Comment:
Isn't something missing here  like `hideRoomsWithNoActivity != undefined`?

Reply:
I had two different deprecation messages based on `hideRoomsWithNoActivity` being undefined or not, but we decided to merge them into one per [this request](https://github.com/RocketChat/Rocket.Chat/pull/32493#discussion_r1638648260)
The idea here is that even users who don't send the `hideRoomsWithNoActivity` param need to be warned that this endpoint's results will be changed in the next major (it will behave as `hideRoomsWithNoActivity=true` by default)

Reply:
Got it, so I'll only call this when the deprecated param is defined, then. However, even users who don't use this param will perceive a breaking change in 7.0

Reply:
Hey @ggazzo
I had a brief discussion with @sampaiodiego about this deprecation message and I believe we agreed that even users who don't use the new param should be warned that this endpoint will be broken on 7.0 (we considered the metrics collected from this function, but this is a relevant message). That's because not only the new param will be removed, but the "default" set of results will also be changed on 7.0. I decided to remove the condition to display the deprecation message so that all users are aware of the incoming change when they use the endpoint
I also considered one more option: not removing the new param at all, but the current behavior can be considered destructive -- so I believe that's something we wouldn't like to keep.

Please tell me what you think.

Reply:
I think no one uses this endpoint so whatever, but it is something non-standard, and does not apply to any other special case, in other scenarios this would flood the logs and create more harm than good. I wouldn't like to see this being done elsewhere. let keep this in mind.

Reply:
Maybe warn if the parameter is NOT used (since our UI will use it by default)